R/gh_response.R

```diff
@@ -1,33 +1,35 @@
-gh_process_response <- function(response) {
-  stopifnot(inherits(response, "response"))
-  if (status_code(response) >= 300) {
-    gh_error(response)
+gh_process_response <- function(resp) {
+  stopifnot(inherits(resp, "httr2_response"))
+  if (httr2::resp_status(resp) >= 300) {
+    gh_error(resp)
   }
 
-  content_type <- http_type(response)
-  gh_media_type <- headers(response)[["x-github-media-type"]]
-  is_raw <- content_type == "application/octet-stream" ||
+  content_type <- httr2::resp_content_type(resp)
+  gh_media_type <- httr2::resp_header(resp, "x-github-media-type")
+
+  is_raw <- identical(content_type, "application/octet-stream") ||
     isTRUE(grepl("param=raw$", gh_media_type, ignore.case = TRUE))
-  is_ondisk <- inherits(response$content, "path")
+  is_ondisk <- inherits(resp$body, "httr2_path")
+
   if (is_ondisk) {
-    res <- response$content
+    res <- as.character(resp$body)
+  } else if (is.na(content_type)) {
+    # empty body
+    res <- list()
```

Comment:
This was previously `NULL`, which I'm pretty sure is incorrect because we add attributes below.

Reply:
Good point. OTOH I am afraid that some packages might rely on this being `NULL`, in particular usethis.

Reply:
Hmmm I thought setting an attribute on `NULL` coerced it to a list, but instead it errors:

``` r
x <- NULL
attr(x, "boo") <- 1
#> Error in attr(x, "boo") <- 1: attempt to set an attribute on NULL
```

<sup>Created on 2023-02-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

So I don't understand why this code currently works.

Reply:
Oh, OK, then I must have remembered something else.

Reply:
All the usethis test pass with this version of gh and I've inspect the usethis code for calls to `DELETE`, and it looks ok so. I'll see if I can come up with a test for a GET call that doesn't return any output (maybe a search for something that doesn't exist) and I'll also installed locally so I can see if anything comes up in interactive use.

Reply:
I think maybe the `PUT`, `POST`, etc. requests return an empty response, no? 

Usethis has some "manual" tests, those are the ones that gh usually breaks. But we can ask Jenny to run them once we are done with the transition to httr2, before the next gh release.

Reply:
I've found three cases that all return empty bodies and added them to the tests.